File: mobile/src/food.js
export const foodGroups = [
  {
    label: '육류',
    items: [
      ['소고기', '🥩'],
      ['국거리용 소고기', '🥩'],
      ['돼지고기', '🍖'],
      ['삼겹살', '🥓'],
      ['닭고기', '🍗'],
      ['달걀', '🥚'],
    ],
  },
  {
    label: '채소',
    items: [
      ['상추', '🥬'],
      ['배추', '🥬'],
      ['당근', '🥕'],
      ['감자', '🥔'],
      ['토마토', '🍅'],
      ['양파', '🧅'],
    ],
  },
  {
    label: '과일',
    items: [
      ['사과', '🍎'],
      ['바나나', '🍌'],
      ['딸기', '🍓'],
      ['포도', '🍇'],
    ],
  },
  {
    label: '유제품/기타',
    items: [
      ['우유', '🥛'],
      ['치즈', '🧀'],
      ['김치', '🥬'],
      ['두부', '⬜'],
    ],
  },
];

export const firstFood = {
  category: '육류',
  subcategory: '육류',
  itemName: '소고기',
  emoji: '🥩',
};

export const unitOptions = ['개', '팩', '봉지', '통', '병', 'g', 'kg', 'ml', 'L'];

export function defaultExpiryDate(days = 7) {
  return new Date(Date.now() + days * 86400000).toISOString().slice(0, 10);
}

File: backend/src/main/java/com/foodmanager/domain/Food.java
package com.foodmanager.domain;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

public class Food {
    private final Long id;
    private final Long sellerId;
    private final String itemName;
    private final Category category;
    private final String subcategory;
    private final String emoji;
    private Integer quantity;
    private final String unit;
    private final BigDecimal price;
    private LocalDate expiryDate;

    public Food(Long id, String itemName, String category, String subcategory, Integer quantity, String unit, LocalDate expiryDate) {
        this(id, null, itemName, Category.fromLabel(category), subcategory, null, quantity, unit, BigDecimal.ZERO, expiryDate);
    }

    public Food(
            Long id,
            Long sellerId,
            String itemName,
            Category category,
            String subcategory,
            String emoji,
            Integer quantity,
            String unit,
            BigDecimal price,
            LocalDate expiryDate
    ) {
        this.id = id;
        this.sellerId = sellerId;
        this.itemName = itemName;
        this.category = category == null ? Category.ETC : category;
        this.subcategory = subcategory;
        this.emoji = emoji;
        this.quantity = quantity;
        this.unit = unit;
        this.price = price == null ? BigDecimal.ZERO : price;
        this.expiryDate = expiryDate;
    }

    public Long id() {
        return id;
    }

    public Long getId() {
        return id;
    }

    public Long sellerId() {
        return sellerId;
    }

    public Long getSellerId() {
        return sellerId;
    }

    public String itemName() {
        return itemName;
    }

    public String getItemName() {
        return itemName;
    }

    public Category category() {
        return category;
    }

    public String getCategory() {
        return category.label();
    }

    public String subcategory() {
        return subcategory;
    }

    public String getSubcategory() {
        return subcategory;
    }

    public String emoji() {
        return emoji;
    }

    public String getEmoji() {
        return emoji;
    }

    public Integer quantity() {
        return quantity;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public String unit() {
        return unit;
    }

    public String getUnit() {
        return unit;
    }

    public BigDecimal price() {
        return price;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public LocalDate expiryDate() {
        return expiryDate;
    }

    public LocalDate getExpiryDate() {
        return expiryDate;
    }

    public void updateQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public void updateExpiryDate(LocalDate expiryDate) {
        this.expiryDate = expiryDate;
    }

    public Integer calculateDaysUntilExpiry(LocalDate today) {
        if (today == null || expiryDate == null) {
            return null;
        }
        return Math.toIntExact(ChronoUnit.DAYS.between(today, expiryDate));
    }

    public Integer calculateDaysUntillExpiry(LocalDate today) {
        return calculateDaysUntilExpiry(today);
    }

    public boolean isExpired(LocalDate today) {
        return today != null && expiryDate != null && expiryDate.isBefore(today);
    }
}
